test/libdnf/rpm/test_package_set.cpp

```diff
@@ -166,20 +166,31 @@ void RpmPackageSetTest::test_iterator() {
     }
 
     // check if begin() points to the first package
-    auto beg = set1->begin();
-    CPPUNIT_ASSERT((*beg).get_id().id == 0);
+    auto it1 = set1->begin();
+    CPPUNIT_ASSERT_EQUAL((*it1).get_id().id, 0);
 
-    // test if end() points to package with id == -2
-    auto end = set1->end();
-    CPPUNIT_ASSERT((*end).get_id().id == -2);
+    // test pre-increment operator
+    auto it2 = ++it1;
+    CPPUNIT_ASSERT_EQUAL((*it1).get_id().id, 1);
+    CPPUNIT_ASSERT_EQUAL((*it2).get_id().id, 1);
 
-    // test if increasing an iterator moves to the second package
-    ++beg;
-    CPPUNIT_ASSERT((*beg).get_id().id == 1);
+    // test post-increment operator
+    auto it3 = it2++;
+    CPPUNIT_ASSERT_EQUAL((*it2).get_id().id, 2);
+    CPPUNIT_ASSERT_EQUAL((*it3).get_id().id, 1);
+
+    // test begin()
+    it1.begin();
+    CPPUNIT_ASSERT_EQUAL((*it1).get_id().id, 0);
+    CPPUNIT_ASSERT(it1 == set1->begin());
+
+    // test end()
+    it1.end();
+    CPPUNIT_ASSERT(it1 == set1->end());
 
     // test iterating the whole package set
-    for (auto it = set1->begin(); it != set1->end(); it++) {
```

Comment:
Hwat about to test both way of iteration? `for (auto it = set1->begin(); it != set1->end(); it++)` and `for (auto pkg : *set1)`.

Reply:
OK. I will implement test loop with pre-increment and with post-increment operator for all iterators.
`for(auto it = set1->begin(), end = set1->end(); it != end; ++it)` and `for(auto it = set1->begin(), end = set1->end(); it != end; it++)`.

Notation `for (auto pkg : *set1)` produces the same iterators calls as the pre-increment version of the test.
https://en.cppreference.com/w/cpp/language/range-for